dspace-iiif/src/main/java/org/dspace/app/iiif/service/WordHighlightSolrSearch.java

```diff
@@ -225,13 +234,16 @@ private AnnotationGenerator getAnnotation(JsonNode highlight, String pageId, UUI
      * canvas identifier in the manifest. For METS/ALTO documents, the page
      * order can be derived from the METS file when loading the solr index.
      * @param pagesNode the pages node
-     * @return canvas id
+     * @return canvas id or null if node was null
      */
     private String getCanvasId(JsonNode pagesNode) {
-        JsonNode page = pagesNode.get(0);
-        String[] identArr = page.get("id").asText().split("\\.");
-        // the canvas id.
-        return "c" + identArr[1];
+        if (pagesNode != null) {
+            JsonNode page = pagesNode.get(0);
+            String[] identArr = page.get("id").asText().split("\\.");
```

Comment:
The previous line's `get` could return `null`, so `page` could be `null` at this point.

What a pain that `get` doesn't return an empty JsonNode.

Reply:
Come to think of it, line 242's `get` could also return `null`.

Reply:
@mwoodiupui : Ok, this is fixed.  There were two methods that used `get()` calls from JsonNode.  Patched them both in latest commit.